File: user_python/Foo_Bar.py
#
# FooBar module
#

import subprocess
from os.path import dirname

fb_path = dirname(dirname(__file__)) + "/user_php"

########################
# get value
#
def foo(one):
    global fb_path
    proc = subprocess.Popen(["php-cli", fb_path+"/Foo_Bar_w.php", one],
        stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
    output, err = proc.communicate()
    if proc.returncode == 0:
        # split on newline
        vals = output.split("\n")
        three = vals[0]
        two = vals[1]
    else:
        three = "failed"
        two = "failed"
    return(three, two)

########################
# set value
#
def bar(three, two, one):
    global fb_path
    proc = subprocess.Popen(["php-cli", fb_path+"/Foo_Bar_w.php", three, two, one],
        stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
    output, err = proc.communicate()
    return proc.returncode

File: user_js/Foo_Bar.js
// this is a javascript wrapper for Foo::Bar Perl module
// server side only (nodejs)

var exec    = require('child_process').execSync;
var process = require('process');
var path    = require('path');

exports.foo = foo;
exports.bar = bar;

var fb_path = path.dirname(path.dirname( __filename )) + "/user_php";

function foo(one) {
  var two;
  var three;
  // call script
  var cmd  = "php-cli "+fb_path+"/Foo_Bar_w.php '"+one+"'";
  var opts = {};
  opts.env = process.env;
  var result = exec(cmd, opts);
  if(result) {
    // why an array of char codes?
    var result_str = "";
    for(var i=0; i< result.length; i++) {
      result_str += String.fromCharCode(result[i]);
    }
    // split on newline
    var output = result_str.split("\n");
    three = output[0];
    two   = output[1];
  } else {
    three = "failed";
    two   = "failed";
    console.log("foo() error");
  }
  return([three,two]);  
}

function bar(three,two,one) {
  // call script
  var cmd  = "php-cli "+fb_path+"/Foo_Bar_w.php '"+three+"' '"+two+"' '"+one+"'";
  var opts = {};
  opts.env = process.env;
  exec(cmd, opts);
}
